pyang/plugins/tree.py

```diff
@@ -127,16 +127,20 @@ def print_help():
 """)
 
 def emit_tree(ctx, modules, fd, depth, llen, path):
-    for module in modules:
-        printed_header = False
 
-        def print_header():
+    def print_header(module):
+        if not printed_header:
             bstr = ""
             b = module.search_one('belongs-to')
             if b is not None:
                 bstr = " (belongs-to %s)" % b.arg
             fd.write("%s: %s%s\n" % (module.keyword, module.arg, bstr))
-            printed_header = True
+            printed_header.append(None)
+
+    printed_header = []
```

Comment:
why did you make this a list and not a boolean?

Reply:
The print_header function wanted to set print_header = True, since that wouldn't work (in python2; in python3, it could be declared nonlocal in the function), the code ended up setting printed_header = True after each call.

Having a mutable list value, which can easily be set (.append(None)), reset (del [:]), and checked (True if nonempty), is an easy solution that works in all python versions.

Reply:
Right, ok, perfect!